### PR TITLE
Refactor time tracking components for Mobile/Desktop view 

### PR DESCRIPTION
**Pull request by Shruti-Apte**
https://www.notion.so/saeloun/The-date-format-and-label-is-incorrect-for-date-field-on-mobile-responsive-edit-entry-page-6fe5d10c7aed44e488a12b97482ae104?pvs=4
https://www.notion.so/saeloun/Refactor-Time-tracking-component-71050425444d46d8afe956c65b7221c4?pvs=4

What: 
- Refactored and extracted Header component
- Started using the same at different places
- Extracted out common usage of the header

Why:
- More readability of the code
- To share common code

Preview: 
Screenshot 2023-03-02 at 6 21 21 PM
Screenshot 2023-03-02 at 6 21 47 PM

Screenshot 2023-03-03 at 3 33 51 PM

### CODE REVIEW

File: app/javascript/src/components/TimeTracking/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+          </span>
+          <span>{currentYear}</span>
+        </>
+      );
+    }
+    if (view === "day") {
+      return isDesktop ? (
+        <>
+          <span>
+            {parseInt(dayInfo[0]["date"], 10)} {dayInfo[0]["month"]} -
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/TimeTracking/Header.tsx:85]: Please add a comment on what functionality this code is handling. Add annotations.

File: app/javascript/src/components/TimeTracking/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      handlePrevWeek();
+    }
+  };
+
+  const getLabel = () => {
+    if (view === "month") {
+      return (
+        <>
+          <span className="mr-2">
+            {monthsAbbr[Math.abs(currentMonthNumber)]}
```
Comment (Shruti-Apte) [inline app/javascript/src/components/TimeTracking/Header.tsx:75]: This line will return the selected month.

File: app/javascript/src/components/TimeTracking/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      );
+    }
+    if (view === "day") {
+      return isDesktop ? (
+        <>
+          <span>
+            {parseInt(dayInfo[0]["date"], 10)} {dayInfo[0]["month"]} -
+          </span>
+          <span className="mx-1">
+            {parseInt(dayInfo[6]["date"], 10)} {dayInfo[6]["month"]}
```
Comment (Shruti-Apte) [inline app/javascript/src/components/TimeTracking/Header.tsx:88]: This component will return the week range of the selected date.

File: app/javascript/src/components/TimeTracking/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+          </span>
+          <span> {dayInfo[6]["year"]}</span>
+        </>
+      ) : (
+        <>
```
Comment (Shruti-Apte) [inline app/javascript/src/components/TimeTracking/Header.tsx:93]: This component will return the selected date on mobile view.

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**